### PR TITLE
test(performance): setup

### PR DESCRIPTION
**Pull request by JAdshead**
## Description

Initial performance test suite which provides a small selection of basic performance tests.

Currently performance tests will need to be run manually and Grafana snapshots published to http://snapshot.raintank.io/

In the near future a "performance test orchestrator"  can be build. 

Example smoke test results - https://snapshot.raintank.io/dashboard/snapshot/jLfY8s4yWzjowpy32r7SaNlm52oxC4vT

## Motivation and Context

Enable basic performance testing locally.

## How Has This Been Tested?

Locally against one-app in development and against production using the prod-sample setup
.
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [x] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

Can use to get a basic idea of one-app performance

### CODE REVIEW

**Comment by mtomcal** [comment]
```
$ docker-compose -f docker-compose.yml -f docker-compose.prod-sample.yml up --abort-on-container-exit influxdb grafana prometheus
ERROR: Network prod-sample_one-app-at-test-network declared as external, but could not be found. Please create the network manually using `docker network create prod-sample_one-app-at-test-network` and try again.
```
Can we have a script that make this super easy and painless to create the network and run docker compose?  Should be one command set up after running prod sample.

**Comment by mtomcal** [comment]
Default date on the Grafana dashboard on first load has Aug 5th baked into the date range. Any way to make it default to current timestamp on first load?

**Comment by JAdshead** [comment]
> ```
> $ docker-compose -f docker-compose.yml -f docker-compose.prod-sample.yml up --abort-on-container-exit influxdb grafana prometheus
> ERROR: Network prod-sample_one-app-at-test-network declared as external, but could not be found. Please create the network manually using `docker network create prod-sample_one-app-at-test-network` and try again.
> ```
> 
> Can we have a script that make this super easy and painless to create the network and run docker compose? Should be one command set up after running prod sample.

This looks i will need to update the readme. when running against prod-sample you will need to ensure the one-app prod-sample server has started before starting the performance monitoring tools. do that will ensure that the prod-sample server has created the external network. There is a larger effort which will be undertaken in the future for writing scripts to abstract this process.

File: __performance__/README.md [hunk 1]
```diff
@@ -67,12 +67,14 @@ The following paths will be expected to be available:
...
 ### Setup
 
-From within the one-app directory start the one-app prod-sample:
+From within the one-app directory first start the one-app prod-sample:
 
 ```bash
 npm run start:prod-sample
 ```
 
+> `npn run start:prod-sample` creates the `prod-sample_one-app-at-test-network` network which is required for the performance tools to start when running against prod-sample.
```
Comment (mtomcal) [inline __performance__/README.md:76]:
```suggestion
> `npm run start:prod-sample` creates the `prod-sample_one-app-at-test-network` network which is required for the performance tools to start when running against prod-sample.
```
I ran start:prod-sample first but it didnt create the docker network that the perf docker compose required. Might need to mention the command to create it manually if things fail.

Comment (JAdshead) [reply __performance__/README.md:76]: for now i am happy with the error provided by docker compose, its fairly explanatory. it takes a while for the prod-sample to finish setting up and actually start the network. maybe you jumped the gun as the one-app server start without that network

File: __performance__/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+Smoke - `/scripts/smoke.js`
+Stress - `/scripts/stress.js`
+Load - `/scripts/load.js`
+Soak - `/scripts/soak.js`
+
+To run a test other than the smoke test replace `/scripts/smoke.js` with the corresponding
+file path outlined above.
+
+Once the test has finished running you can use Grafana to share a snapshot
```
Comment (infoxicator) [inline __performance__/README.md:106]: Worth adding the caveat that you have to reload grafana to see the results of the test?

File: __performance__/README.md (hunk 1 above)
Comment (mtomcal) [reply __performance__/README.md]: As long as this is addressed later, Im good with this.

File: __performance__/README.md (hunk 2 above)
Comment (JAdshead) [reply __performance__/README.md:106]: you should not have to, there is a little button at the top right that triggers another poll

File: prod-sample/docker-compose.yml [hunk 3]
```diff
@@ -15,6 +15,8 @@ services:
     expose:
       - "8443"
       - "3005"
+    ports:
+      - "3005:3005"
```
[the remark this answers is not in the document]
Comment (JAdshead) [reply prod-sample/docker-compose.yml:19]: yes, needs to be exposed to gather metrics.

File: __performance__/prometheus/targets.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[
+  {
+    "targets": ["host.docker.internal:3005"],
```
[the remark this answers is not in the document]
Comment (JAdshead) [reply __performance__/prometheus/targets.json:3]: yep, prometheus needs to use the this to scrape the metrics

**Pull request merged**